lib/string_decoder.js

```diff
@@ -14,6 +16,8 @@ function normalizeEncoding(enc) {
   return nenc || enc;
 }
 
+const simpleWrite = {};
```

Comment:
Also, it might be worthwhile comparing other lookup strategies, such as using a function that returns the correct function based on the encoding.